index.html

```diff
@@ -1831,322 +1831,291 @@ <h2>Proof</h2>
   <section class="normative">
     <h1>Core Representations</h1>
 
-      <p>
-A <a>DID document</a> MUST be a single JSON object conforming to [[!RFC8259]].
-Many of the concepts in this document were introduced by example using the
-JSON-LD syntax, a format for mapping JSON data into the RDF semantic graph
-model, as defined by [[!JSON-LD]]. This section formalizes how the data model
-(described in Sections <a href="#data-model"></a> and
-<a href="#did-documents"></a>) is realized in JSON-LD.
-      </p>
-      <p>
-Although syntactic mappings are provided for JSON and JSON-LD only, applications
-and services can use any other data representation syntax, such as JXD (JSON XDI
-Data, a serialization format for the
-<a href="http://docs.oasis-open.org/xdi/xdi-core/v1.0/csd01/xdi-core-v1.0-csd01.xml">XDI graph model</a>),
-XML, YAML, or CBOR, that is capable of expressing the data model.
-      </p>
-
-    <section>
-      <h2>
-Pure JSON
-      </h2>
-
-        <p>
-The data model, as described in Section <a href="#data-model"></a>, can be
-encoded in Javascript Object Notation (JSON) [[RFC8259]] by mapping property
-values to JSON types as follows:
-        </p>
-
-        <ul>
-          <li>
-Numeric values representable as IEEE754 SHOULD be represented as a Number type.
-          </li>
-          <li>
-Boolean values SHOULD be represented as a Boolean type.
-          </li>
-          <li>
-Sequence value SHOULD be represented as an Array type.
-          </li>
-          <li>
-Unordered sets of values SHOULD be represented as an Array type.
-          </li>
-          <li>
-Sets of properties SHOULD be represented as an Object type.
-          </li>
-          <li>
-Empty values SHOULD be represented as a null value.
-          </li>
-          <li>
-Other values MUST be represented as a String type.
-          </li>
-        </ul>
-    </section>
+    <p>
+All concrete representations of a <a>DID document</a> MUST be serialized using a 
+deterministic mapping that is able to be unambiguously parsed into the data model
+defined in this specification. All serialization methods MUST define rules for 
+the bidirectional translation of a <a>DID document</a> both into and out of the
+representation in question. As a consequence, translation between any two representations
+MUST be done by parsing the source format into a <a>DID document</a> model
```

Comment:
Untestable normative requirement? Struggling to understand how we write a test for this normative statement.

Reply:
Not every requirement can be tested automatically. That doesn't make the requirement meaningless.

Reply:
> Not every requirement can be tested automatically. That doesn't make the requirement meaningless.

Over the last decade or so, W3C WGs have, in general, started to move away from including normative language that isn't testable in a test suite. W3C WGs are going this direction because of historical fighting over what an untestable normative statement actually means and what should be done if an implementation doesn't do the untestable normative thing. By and large, these normative statements are useless. There are exceptions, like normative language about how you should write a specification -- the DID Method section in this spec is a good example of this sort of language... but requiring a human being to step in and interpret whether or not an implementation is conformant is generally regarded as an anti-pattern.

Normative statements that require human intervention don't get put in test suites and thus those statements are effectively useless over time because developers run their implementations against a test suite, and if the test suite passes, the devs move on.

I was seeing if you had a plan about how this statement would be tested, and I'm hearing "we're not going to test this statement in the test suite", which automatically puts it into the "then let's not make it a normative requirement". You could restate it as this:

> translation between any two representations **is expected to** be done by parsing the source format into a <a>DID document</a> model

I suggest "is expected to" because there may be other valid ways to do this, like through some XSLT-like process that can do a direct translation w/o "going into" the DID document model (whatever that means).

I can merge this as is, but I expect that we'll remove it in time because it's not saying anything useful from a testability perspective.